### PR TITLE
Custom Feedback Paths: Visibility

### PR DESCRIPTION
**Pull request by JosephineKwa**
1. Updated checks to allow session and responses to be visible to custom feedback participants and hidden to non-participants where applicable (first 5 commits)
2. Updated email generation to generate emails correctly for custom feedback participants (next 3 commits)

### CODE REVIEW

**Comment by JosephineKwa** [comment]
@chowyb I have made the necessary changes, ready for review

File: src/test/java/teammates/test/cases/action/InstructorCourseDetailsPageActionTest.java [hunk 1]
```diff
@@ -165,6 +165,14 @@ public void testExecuteAndPostProcess() {
...
                              + "<td>student5InCourse1@gmail.tmt</td>"
                          + "</tr>"
+                         + "<tr>"
+                             + "<td>Section 2</td>"
+                             + "<td>Team 1.3</td>"
+                             + "<td>student6 In Course1</td>"
+                             + "<td>Course1</td>"
+                             + "<td>Joined</td>"
+                             + "<td>student6InCourse1@gmail.tmt</td>"
+                         + "</tr>"
```
Comment (chowyb) [inline src/test/java/teammates/test/cases/action/InstructorCourseDetailsPageActionTest.java]: I had no idea a test like this existed 😱

**Comment by JosephineKwa** [comment]
Alright, thanks for the review!

**Comment by damithc** [comment]
@whipermr5 this is a 'long running feature branch'. I'll review whenl the full feature is ready.

**Comment by whipermr5** [comment]
@damithc Oops sorry!! 😅

**Pull request merged**